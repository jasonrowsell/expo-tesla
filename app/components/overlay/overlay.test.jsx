import React from 'react';
import { render } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import { Text } from 'react-native';

import Overlay from './overlay';

describe('<Overlay />', () => {
  const renderComponent = () => render(
    <Overlay open>
      <Text>
        My Content
      </Text>
    </Overlay>,
  );

  test('has 1 child', () => {
    const tree = renderer.create(
      <Overlay open>
        <Text>
          My Content
        </Text>
      </Overlay>,
    ).toJSON();
    expect(tree.children.length).toBe(1);
  });

  test('renders correctly', () => {
    const tree = renderComponent();
    expect(tree).toMatchSnapshot();
  });

  test('renders to the page if open is true', () => {
    const { getByText } = renderComponent();
    getByText('My Content');
  });

  test('does not render to the page if open is false', () => {
    const { queryByText } = render(
      <Overlay open={false}>
        <Text>
          My Content
        </Text>
      </Overlay>,
    );
    expect(queryByText('My Content')).not.toBeTruthy();
  });
});
