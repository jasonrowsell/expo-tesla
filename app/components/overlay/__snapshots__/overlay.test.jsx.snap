// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Overlay /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(0, 0, 0, 0.5)",
      "flex": 1,
      "height": 1334,
      "left": 0,
      "position": "absolute",
      "top": -50,
      "width": 750,
    }
  }
>
  <Text>
    My Content
  </Text>
</View>
`;
